pkg/apis/httpEndpoint/v1alpha1/types.go

```diff
@@ -51,11 +61,61 @@ func (h HTTPEndpoint) NameValuePairs() []common.NameValuePair {
 	return h.Spec.Headers
 }
 
+// nil returns a bool indicating if a client key has a secret reference
+func (h HTTPEndpoint) HasTLSClientKeySecret() bool {
+	return h.Spec.ClientKey != nil && h.Spec.ClientKey.SecretKeyRef != nil && h.Spec.ClientKey.SecretKeyRef.Name != ""
+}
+
+// HasTLSClientCertSecret returns a bool indicating if a client cert has a secret reference
+func (h HTTPEndpoint) HasTLSClientCertSecret() bool {
+	return h.Spec.ClientCert != nil && h.Spec.ClientCert.SecretKeyRef != nil && h.Spec.ClientCert.SecretKeyRef.Name != ""
+}
+
+// HasTLSRootCASecret returns a bool indicating if a root cert has a secret reference
+func (h HTTPEndpoint) HasTLSRootCASecret() bool {
+	return h.Spec.RootCA != nil && h.Spec.RootCA.SecretKeyRef != nil && h.Spec.RootCA.SecretKeyRef.Name != ""
+}
+
+// HasTLSRootCA returns a bool indicating if the HTTP endpoint contains a root ca
+func (h HTTPEndpoint) HasTLSRootCA() bool {
+	return h.Spec.RootCA != nil && h.Spec.RootCA.Value != nil
+}
+
+// HasTLSClientCert returns a bool indicating if the HTTP endpoint contains a client cert
+func (h HTTPEndpoint) HasTLSClientCert() bool {
+	return h.Spec.ClientCert != nil && h.Spec.ClientCert.Value != nil
+}
+
+// HasTLSClientKey returns a bool indicating if the HTTP endpoint contains a client key
+func (h HTTPEndpoint) HasTLSClientKey() bool {
+	return h.Spec.ClientKey != nil && h.Spec.ClientKey.Value != nil
+}
+
+// TLS describes tls specific properties for endpoint authenticatin
+type TLS struct {
+	// Value of the property, in plaintext.
+	//+optional
+	Value *common.DynamicValue `json:"value,omitempty"`
+	// SecretKeyRef is the reference of a value in a secret store component.
+	//+optional
+	SecretKeyRef *common.SecretKeyRef `json:"secretKeyRef,omitempty"`
+}
+
 // HTTPEndpointSpec describes an access specification for allowing external service invocations.
 type HTTPEndpointSpec struct {
 	BaseURL string `json:"baseUrl" validate:"required"`
 	//+optional
 	Headers []common.NameValuePair `json:"headers"`
+	//+optional
+	RootCA *TLS `json:"tlsRootCA"`
+	//+optional
+	ClientCert *TLS `json:"tlsClientCert"`
+	//+optional
+	ClientKey *TLS `json:"tlsClientKey"`
+	//+optional
+	//+kubebuilder:validation:Enum={"Never", "OnceAsClient", "FreelyAsClient"}
+	//+default=Never
```

Comment:
My bad, I think the tag also needs the `kubebuilder` prefix. We should see the `default` field in the generated CRD yaml for this type.

```suggestion
	// +kubebuilder:default=Never
```